lib/parse.js

```diff
@@ -1,87 +1,73 @@
 "use strict";
 
 var babylonToEspree = require("./babylon-to-espree");
-var parse = require("babylon").parse;
 var tt = require("babylon").tokTypes;
 var traverse = require("@babel/traverse").default;
 var codeFrameColumns = require("@babel/code-frame").codeFrameColumns;
 
+var ESLINT_DEFAULT_FILEPATH = "<text>";
+
 module.exports = function(code, options) {
   var opts = {
-    codeFrame: options.hasOwnProperty("codeFrame") ? options.codeFrame : true,
-    sourceType: options.sourceType,
-    allowImportExportEverywhere: options.allowImportExportEverywhere, // consistent with espree
-    allowReturnOutsideFunction: true,
-    allowSuperOutsideMethod: true,
-    ranges: true,
-    tokens: true,
-    plugins: [
-      "flow",
-      "jsx",
-      "estree",
-      "asyncFunctions",
-      "asyncGenerators",
-      "classConstructorCall",
-      "classProperties",
-      "decorators",
-      "doExpressions",
-      "exponentiationOperator",
-      "exportDefaultFrom",
-      "exportNamespaceFrom",
-      "functionBind",
-      "functionSent",
-      "objectRestSpread",
-      "trailingFunctionCommas",
-      "dynamicImport",
-      "numericSeparator",
-      "optionalChaining",
-      "importMeta",
-      "classPrivateProperties",
-      "bigInt",
-      "optionalCatchBinding",
-      "throwExpressions",
-      "pipelineOperator",
-      "nullishCoalescingOperator",
-    ],
+    filename: (options.filePath && options.filePath !== ESLINT_DEFAULT_FILEPATH) ? options.filePath : '',
+    parserOpts: {
+      sourceType: options.sourceType,
+      allowImportExportEverywhere: options.allowImportExportEverywhere, // consistent with Espree
+      allowReturnOutsideFunction: true,
+      allowSuperOutsideMethod: true,
+      ranges: true,
+      tokens: true,
+    },
   };
+  var enableCodeFrame = options.hasOwnProperty("codeFrame") ? options.codeFrame : true;
```

Comment:
This might break if i pass a “hasOwnProperty” option; you probably want `Object.prototype.hasOwnProperty.call` or the `has` module (i realize this isn’t related to your PR, but it’s an easy fix)